plugins/modules/archive.py

```diff
@@ -542,6 +550,33 @@ def _get_checksums(self, path):
         return checksums
 
 
+class ReproducibleTGZFile(tarfile.TarFile):
+    def __init__(
+        self, name=None, mode=None, compresslevel=-1, fileobj=None, mtime=None, **kwargs
+    ):
+        if fileobj is None:
+            fileobj = open(name, mode + "b")
+
+        try:
+            # output filename intentionally empty exclude it from gzip header
+            gzipfileobj = gzip.GzipFile("", mode, compresslevel, fileobj, mtime)
+        except Exception:
+            fileobj.close()
+            raise
+
+        # Allow GzipFile to close fileobj as needed
+        gzipfileobj.myfileobj = fileobj
```

Comment:
Hmm, this is not officially documented (https://docs.python.org/3/library/gzip.html#gzip.GzipFile), so this looks like it's relying on an implementation detail of GzipFile to me.